volttron/utils/rmq_setup.py

```diff
@@ -168,7 +168,48 @@ def write_env_file(rmq_config, conf_file, env=None):
             env_conf.write(env_entries)
 
 
-def _create_federation_setup(admin_user, admin_password, is_ssl, vhost, vhome):
+def _get_federation_certs(vhome):
+    federation_config_file = os.path.join(vhome,
+                                          'rabbitmq_federation_config.yml')
+    federation_config = read_config_file(federation_config_file)
+    federation = federation_config.get('federation-upstream', {})
+    success = True
+    update_needed = False
+    try:
+        for host, upstream in federation.items():
+            rmq_user = 'federation' #upstream['federation-user']
+            if 'certificates' not in upstream:
+                # certificates key not found in shovel config
+                https_port = upstream.get('https_port', 8443)
+                remote_addr = 'https://{}:{}'.format(host, https_port)
+                print(f"Certificates not found. Requesting CSR from {remote_addr}")
+                # request CSR from remote host
+                ca_file, cert_file, prvt_file = _request_csr(rmq_user, remote_addr, 'federation')
+                if ca_file is not None and cert_file is not None and prvt_file is not None:
+                    upstream['certificates'] = {}
+                    # root CA
+                    upstream['certificates']['remote_ca'] = ca_file
+                    # public cert
+                    upstream['certificates']['public_cert'] = cert_file
+                    # private_key
+                    upstream['certificates']['private_key'] = prvt_file
+                    update_needed = True
+                else:
+                    _log.error(
+                        f"ERROR: Couldn't get CSR certificates from remote server. Check with admin of remote instance."
+                        "\nContinuing with other configurations")
```

Comment:
When a csr is denied I get the message "continuing with other configurations" twice even though there is no more federation configuration in the file. File had only one entry. Looks like _get_federation_certs is returning True even when csr is denied and hence _create_federation_setup is called which throws "certificate key missing"

(volttron) volttron@volttron1:~/git/myvolttron$ vcfg rabbitmq federation

Your VOLTTRON_HOME currently set to: /home/volttron/fed_v1

Is this the volttron you are attempting to setup? [Y]:
Name of this volttron instance: [volttron1]:
rabbitmq_federation_config.yml exists in /home/volttron/fed_v1/rabbitmq_federation_config.yml Do you wish to use this file to configure federation [Y]:
Certificates not found. Requesting CSR from https://volttron2:8443
INFO: Request for signed certificate(CSR) has been denied by the remote instance administrator
ERROR: ERROR: Couldn't get CSR certificates from remote server. Check with admin of remote instance.
Continuing with other configurations
ERROR: ERROR: Certificates key missing in config.
Continuing with other configurations